### PR TITLE
Added clip function to utilities and implemented

### LINKED ISSUE #24: Add A Saturate Or Clip Utility Function
Takes An Input, An Upper Bound, And A Lower Bound, Clips The Input To The Boundary If It Is Not Between The Bounds
Also implement this in any subsystems or commands that need it,

### PR DESCRIPTION
**Pull request by MatthewZ0823**
Fix #24 

Added a clip function that takes an input, an upper bound, and a lower bound, clips the input to the boundary if it's not between the bounds

### CODE REVIEW

File: src/main/java/frc/robot/util/Utilities.java [hunk 1]
```diff
@@ -27,4 +27,8 @@ public static double deadzone(double input) {
   public static double deadzone(double input, double deadzone) {
     return Math.abs(input) < Math.abs(deadzone) ? 0 : input;
   }
+
+  public static double clip(double input, double max, double min) {
```
Comment (Jacob4649) [inline src/main/java/frc/robot/util/Utilities.java]: Nitpick: Change the order of the parameters to `double input, double min, double max`, it just makes more sense intuitively

**Review by Jacob4649 — CHANGES_REQUESTED** [review]
Make sure you implement this new method everywhere necessary, I could be wrong, but it seems as though you're missing a few spots right now

File: src/main/java/frc/robot/util/Utilities.java (hunk 1 above)
Comment (matthew-reynolds) [reply src/main/java/frc/robot/util/Utilities.java]: Resolved in 90cb14b

File: src/main/java/frc/robot/util/Utilities.java [hunk 2]
```diff
@@ -27,4 +27,8 @@ public static double deadzone(double input) {
   public static double deadzone(double input, double deadzone) {
     return Math.abs(input) < Math.abs(deadzone) ? 0 : input;
   }
+
+  public static double clip(double input, double min, double max) {
```
Comment (matthew-reynolds) [inline src/main/java/frc/robot/util/Utilities.java:39]: Please add a quick javadoc here

**Review by matthew-reynolds — CHANGES_REQUESTED** [review]
Looks good, although like @Jacob4649 said let's double check that we're using this new utility everywhere applicable. Thanks!

**Comment by matthew-reynolds** [comment]
Did a quick search, from what I can see the only missing place we should use this new utility is in `AutoAlignTurret.java`, similar to your change in `MoveTurret.java`.

**Comment by MatthewZ0823** [comment]
> Make sure you implement this new method everywhere necessary, I could be wrong, but it seems as though you're missing a few spots right now

Resolved in commit c1e721d

File: src/main/java/frc/robot/commands/turret/AutoAlignTurret.java [hunk 3]
```diff
@@ -23,16 +24,15 @@
...
         () -> {
           double angle = (turret.getAngle() + limelight.getAngleOffset());
-          angle = Math.max(Constants.TURRET_ROTATION_MIN, angle);
-          angle = Math.min(Constants.TURRET_ROTATION_MAX, angle);
+          Utilities.clip(angle, Constants.TURRET_ROTATION_MIN, Constants.TURRET_ROTATION_MAX);
 
           return angle;
-        }, 
-        
+        },
```
Comment (matthew-reynolds) [inline src/main/java/frc/robot/commands/turret/AutoAlignTurret.java]: Can be simplified to a single line:

```
() -> Utilities.clip(turrent.getAngle() + limelight.getAngleOffset(), Constants.TURRET_ROTATION_MIN, Constants.TURRET_ROTATION_MAX),
```

File: src/main/java/frc/robot/util/Utilities.java (hunk 2 above)
Comment (MatthewZ0823) [reply src/main/java/frc/robot/util/Utilities.java:39]: Resolved in commit 2477658

**Review by matthew-reynolds — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/commands/turret/AutoAlignTurret.java (hunk 3 above)
Comment (MatthewZ0823) [reply src/main/java/frc/robot/commands/turret/AutoAlignTurret.java]: Resolved in commit cea377d

**Review by matthew-reynolds — APPROVED** [review]
Thanks!

**Pull request merged**